### PR TITLE
Update version of zig

### PR DESCRIPTION
**Pull request by wrongnull**
Zig has a newer version

### CODE REVIEW

**Comment by TomJo2000** [comment]
The update to 0.10.0 got stalled a couple months ago, I'm not sure those issues were ever resolved #12720

File: packages/zig/build.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@ TERMUX_PKG_HOMEPAGE=https://ziglang.org
 TERMUX_PKG_DESCRIPTION="General-purpose programming language and toolchain for maintaining robust, optimal, and reusable software."
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@leapofazzam123"
-TERMUX_PKG_VERSION=0.9.1
+TERMUX_PKG_VERSION=0.10.1
 TERMUX_PKG_SRCURL=https://ziglang.org/download/$TERMUX_PKG_VERSION/zig-$TERMUX_PKG_VERSION.tar.xz
 TERMUX_PKG_SHA256=38cf4e84481f5facc766ba72783e7462e08d6d29a5d47e3b75c8ee3142485210
```
Comment (TomJo2000) [inline packages/zig/build.sh:7]: Please remember to update the SHA256 hash.

**Review by TomJo2000 — COMMENTED** [review]
SHA256 sum not updated.

**Comment by xtkoba** [comment]
If you are going to do trial and error on CI, then please submit another PR that you believe is trivial. Please do not bother us running checks manually.

**Comment by wrongnull** [comment]
> The update to 0.10.0 got stalled a couple months ago, I'm not sure those issues were ever resolved #12720

what kind of errors do we get there?

**Comment by TomJo2000** [comment]
I don't think the CI logs are available anymore unfortunately.

**Comment by buttaface** [comment]
I tried this locally after updating this patch slightly a couple weeks ago, but it wouldn't build against LLVM 16:
```
diff --git a/packages/zig/cmake-Findllvm.cmake.patch b/packages/zig/cmake-Findllvm.cmake.patch
index c42914804..28c678117 100644
--- a/packages/zig/cmake-Findllvm.cmake.patch
+++ b/packages/zig/cmake-Findllvm.cmake.patch
@@ -6,6 +6,6 @@
    FIND_AND_ADD_LLVM_LIB(LLVMDemangle)
 +  FIND_AND_ADD_LLVM_LIB(Polly)
 +  FIND_AND_ADD_LLVM_LIB(PollyISL)
- endif()

- include(FindPackageHandleStandardArgs)
+   get_filename_component(LLVM_LIBDIRS "${LLVM_LLVMCORE_LIB}" DIRECTORY)
+   find_path(LLVM_INCLUDE_DIRS NAMES llvm/IR/IRBuilder.h)
```
More work is required to get Zig building again.

**Pull request closed without merging**